lib/browser/tag/each.js

```diff
@@ -170,79 +132,76 @@ export default function _each(dom, parent, expr) {
           item
         }, dom.innerHTML)
 
+        // link the original item to it's tag
+        if (isObject(_item))
+          defineProperty(_item, '_riot_id', tag._riot_id)
+
+        tagsMounted[tag._riot_id] = tag
+
         tag.mount()
-        domToInsert = tag.root
-        // this tag must be appended
-        if (i == tags.length) {
-          if (isVirtual)
-            makeVirtual(tag, frag)
-          else frag.appendChild(domToInsert)
-        }
-        // this tag must be insert
-        else {
-          if (isVirtual)
-            makeVirtual(tag, root, tags[i])
-          else root.insertBefore(domToInsert, tags[i].root)
-          oldItems.splice(i, 0, item)
-        }
 
-        tags.splice(i, 0, tag)
-        if (child) arrayishAdd(parent.tags, tagName, tag, true)
-        pos = i // handled here so no move
-      } else tag.update(item)
-
-      // reorder the tag if it's not located in its previous position
-      if (pos !== i && _mustReorder) {
-        // update the DOM
-        if (isVirtual)
-          moveVirtual(tag, root, tags[i], dom.childNodes.length)
```

Comment:
Did you follow the link and swap?  The whole idea of moving elements is to allow third party libraries to change the dom and not be affected with updates(user input should be preserved also).  The example I linked should keep the color which is a simulation of a third party dom manipulation.